workitem/workitem_repository.go

```diff
@@ -960,3 +961,38 @@ func (r *GormWorkItemRepository) GetCountsForIteration(ctx context.Context, itr
 	}
 	return countsMap, nil
 }
+
+// LoadByIteration returns the list of work items belongs to given iteration
+func (r *GormWorkItemRepository) LoadByIteration(ctx context.Context, iterationID uuid.UUID) ([]*WorkItem, error) {
+	defer goa.MeasureSince([]string{"goa", "db", "workitem", "loadByIteration"}, time.Now())
+	log.Info(nil, map[string]interface{}{
+		"itr_id": iterationID,
+	}, "Loading work items for iteration")
+
+	res := []WorkItemStorage{}
+	filter := fmt.Sprintf(`fields @> '{"%s":"%s"}'`, SystemIteration, iterationID)
+	tx := r.db.Model(WorkItemStorage{}).Where(filter).Find(&res)
+	if tx.Error != nil {
+		return nil, errors.NewInternalError(ctx, tx.Error)
+	}
+	workitems := []*WorkItem{}
+	for _, ele := range res {
+		wiType, err := r.witr.LoadTypeFromDB(ctx, ele.Type)
+		if err != nil {
+			log.Error(nil, map[string]interface{}{
+				"wit_id": ele.Type,
+				"err":    err,
+			}, "error in loading type from DB")
+			return nil, errors.NewInternalError(ctx, err)
+		}
+		convertedWI, err := ConvertWorkItemStorageToModel(wiType, &ele)
+		if err != nil {
+			log.Error(nil, map[string]interface{}{
+				"wi_id": ele.ID,
+				"err":   err,
+			}, "error in converting WI")
```

Comment:
Why only log the error but not return it?

Reply:
true that! returned error in https://github.com/fabric8-services/fabric8-wit/pull/1683/commits/8580492244f892db08224fe283e57c8aef620f76